nettrine/app/Presenters/AdvancedPresenter.php

```diff
@@ -0,0 +1,191 @@
+<?php declare(strict_types=1);
+
+namespace App\Presenters;
+
+use App\Model\Database\Advanced\Entity\Article;
+use App\Model\Database\Advanced\Entity\ArticleCategory;
+use App\Model\Database\EntityManagerDecorator;
+use Doctrine\Common\Collections\Criteria;
+use Exception;
+use Gedmo\Loggable\Entity\LogEntry;
+use Nette\Application\UI\Form;
+use Nette\Application\UI\Presenter;
+use Nette\Localization\ITranslator;
+
+class AdvancedPresenter extends Presenter
+{
+
+	/** @var EntityManagerDecorator @inject */
+	public $em;
+
+	/** @var ITranslator @inject */
+	public $translator;
+
+	/** @persistent */
+	public $locale;
+
+	public function renderDefault(?string $locale): void
+	{
+		if (!$locale) {
+			$this->redirect('Advanced:', ['locale' => 'en_GB']);
+		}
+
+		$articleCategoryRepository = $this->em->getArticleCategoryRepository();
+		$articleRepository = $this->em->getArticleRepository();
+
+		$this->template->categories = $articleCategoryRepository->findAllOrderedCategories();
+
+		$articles = [];
+
+		$criteria = new Criteria();
+		$criteria->where(Criteria::expr()->isNull('deletedAt'));
+
+		foreach ($articleRepository->matching($criteria) as $article) {
+			/** @var Article $article */
+			$article->setTranslatableLocale($locale);
+			$this->em->refresh($article);
+
+			$articles[] = $article;
+		}
+
+		$this->template->articles = $articles;
+
+		$articlesHistory = [];
+		foreach ($articles as $article) {
+			/** @var Article $article */
+			$repo = $this->em->getRepository(LogEntry::class); // we use default log entry class
+			$logs = $repo->getLogEntries($article);
+
+			foreach ($logs as $log) {
+				/** @var LogEntry $log */
+				$articlesHistory[$log->getId()]['article'] = $article;
+				$articlesHistory[$log->getId()]['history'] = $log;
+			}
+		}
+
+		ksort($articlesHistory);
+
+		$this->template->articlesHistory = $articlesHistory;
+	}
+
+	protected function createComponentAddArticleCategoryForm(): Form
+	{
+		$form = new Form();
+
+		$form->addText('title', 'messages.article_categories.title')
+			->setRequired('messages.article_categories.title_required');
+
+		$articleCategoryRepository = $this->em->getArticleCategoryRepository();
+		$categories = $articleCategoryRepository->findPairs();
+		$form->addSelect('parent', 'messages.article_categories.parent_category', $categories)
```

Comment:
Nestacilo by jenom `$this->em->getArticleCategoryRepository()->findPairs()` bez prirazovani do promennych?